packages/@lightningjs/ui-components/src/components/ProgressBar/ProgressBar.js

```diff
@@ -59,6 +59,9 @@ export default class ProgressBar extends Base {
   }
 
   _updateTextures() {
+    const p = this.w * this._progress;
+    const w = p <= 0 ? 0 : Math.min(p, this.w);
```

Comment:
Since these same two lines are used farther down in the updateProgress method, how about we make this it's own method, maybe an assessor like `get _progressWidth`?